File: event-client/src/app/recommendation-list/model/EventTopDTO.ts
import {TicketOptionDTO} from '../../search/model/TicketOptionDTO';

export interface EventTopDTO {
  id: number;
  rawText: string;
  eventType: string;
  eventDateTime: string;
  city: string;
  url: string;
  ticketOptions: TicketOptionDTO[];
}

File: src/main/java/com/example/searchingevents/models/dto/EventTopDTO.java
package com.example.searchingevents.models.dto;

import java.util.List;

public record EventTopDTO(Long id,
                          String rawText,
                          String eventType,
                          String eventDateTime,
                          String city,
                          String url,
                          List<TicketOptionDTO> ticketOptions) {
}
